aws/data_source_aws_launch_template.go

```diff
@@ -349,29 +349,10 @@ func dataSourceAwsLaunchTemplateRead(d *schema.ResourceData, meta interface{}) e
 		LaunchTemplateNames: []*string{aws.String(d.Get("name").(string))},
 	})
 
-	if isAWSErr(err, ec2.LaunchTemplateErrorCodeLaunchTemplateIdDoesNotExist, "") {
-		log.Printf("[WARN] launch template (%s) not found - removing from state", d.Id())
-		d.SetId("")
-		return nil
-	}
-
-	// AWS SDK constant above is currently incorrect
-	if isAWSErr(err, "InvalidLaunchTemplateId.NotFound", "") {
-		log.Printf("[WARN] launch template (%s) not found - removing from state", d.Id())
-		d.SetId("")
-		return nil
-	}
-
 	if err != nil {
 		return fmt.Errorf("Error getting launch template: %s", err)
 	}
```

Comment:
let's keep this `if` check to catch and return this error type in addition to `"InvalidLaunchTemplateName.NotFoundException"` if you don't mind adding it in as well since a search by `name` can result in this type of error :) and we can join these if conditions with e.g.
```go
if err != nil {
                if isAWSErr(err, "InvalidLaunchTemplateId.NotFound", "") ||
			isAWSErr(err, "InvalidLaunchTemplateName.NotFoundException", "") {
			return errors.New("Launch Template not found")
		}
		return fmt.Errorf("error reading launch template: %w", err)
}
```